hts-precompile/IHederaTokenService.sol

```diff
@@ -395,7 +395,7 @@ interface IHederaTokenService {
     /// @return tokenAddress the created token's address
     function createFungibleToken(
         HederaToken memory token,
-        uint initialTotalSupply,
+        uint64 initialTotalSupply,
         uint decimals
```

Comment:
decimals is `uint32` - https://github.com/hashgraph/hedera-protobufs/blob/d5eaa07c41999fe8e6611b552d50373f8a809a07/services/token_create.proto#L92

How far the rabbit hole do we go?

Reply:
It makes sense to me to fix it all while I'm here.  There should be no difference in the services code except the hash that is generated.

Reply:
Alongside the hash update, I think we can also [delete the `initSupply` and `decimals` checks in TokenCreatePrecompile](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/TokenCreatePrecompile.java#L284-L297), since they will no longer be needed after this type fix.

Reply:
@dimitar-dinev I believe we might want to keep the checks as the we will still be supporting the old version of these types.